Src/AutoFakeItEasy2.UnitTest/FakeItEasyBuilderTest.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using FakeItEasy;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.TestTypeFoundation;
+using Xunit;
+using Xunit.Extensions;
+
+namespace Ploeh.AutoFixture.AutoFakeItEasy2.UnitTest
+{
+  public class FakeItEasyBuilderTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilder()
+        {
+            // Fixture setup
+            var dummyBuilder = A.Fake<ISpecimenBuilder>();
+            // Exercise system
+            var sut = new FakeItEasyBuilder(dummyBuilder);
+            // Verify outcome
+            Assert.IsAssignableFrom<ISpecimenBuilder>(sut);
+            // Teardown
+        }
+
+        [Fact]
+        public void InitializeWithNullBuilderThrows()
+        {
+            // Fixture setup
+            // Exercise system and verify outcome
+            Assert.Throws<ArgumentNullException>(() =>
+                new FakeItEasyBuilder(null));
+            // Teardown
+        }
+
+        [Fact]
+        public void BuilderIsCorrect()
+        {
+            // Fixture setup
+            var expectedBuilder = A.Fake<ISpecimenBuilder>();
+            var sut = new FakeItEasyBuilder(expectedBuilder);
+            // Exercise system
+            ISpecimenBuilder result = sut.Builder;
+            // Verify outcome
+            Assert.Equal(expectedBuilder, result);
+            // Teardown
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(1)]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(string))]
+        public void CreateWithNonFakeRequestReturnsCorrectResult(object request)
+        {
+            // Fixture setup
+            var dummyBuilder = A.Fake<ISpecimenBuilder>();
+            var sut = new FakeItEasyBuilder(dummyBuilder);
+            // Exercise system
+            var dummyContext = A.Fake<ISpecimenContext>();
+            var result = sut.Create(request, dummyContext);
+            // Verify outcome
+#pragma warning disable 618
```

Comment:
These warnings are there in the existing AutoFixture code base because we were concerned that it'd be a breaking change to shift to returning `new NoSpecimen()`. This library, being new, however, doesn't need to make that concession, so ought to use `new NoSpecimen()` directly, so that the `#pragma` lines can be avoided.